pages/api/contactMentee.js

```diff
@@ -0,0 +1,47 @@
+import { getSession } from "next-auth/react";
+import { getUserById, getUser } from "../../services";
+
+export default async function handler(req, res) {
+  const session = await getSession({ req });
+  if (!session) {
+    res.status(403).json({
+      error: "You must be signed in to access this API.",
+    });
+    return;
+  }
+
+  if (req.method === "POST") {
+    return handlePOST(session, req, res);
+  } else {
+    res
+      .status(404)
+      .json({ message: `Unsupported method on this endpoint: ${req.method}` });
+  }
+}
+
+async function handlePOST(session, req, res) {
+  const user = await getUser(session);
+  if (!user) {
+    res.status(500).json({ message: "Can't get user data" });
+    return;
+  }
+
+  const mentee = await getUserById(req.body.menteeId);
```

Comment:
I wonder if it's worth checking here if the mentee is actually in a relationship with the mentor, otherwise a mentor could contact all mentees by guessing their IDs. Unlikely to happen in our case, so totally happy with leaving this for a follow up P.

Reply:
Totally right, I had time to fix it so I did replace mentee with request, and check if the current user is the mentor for that request id. Thanks for the feedback!

Reply:
Please let me know how this looks, and I'll merge